packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,843 @@
+#!/bin/sh
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# ======================================================================
+# Constants
+
+REPOCONFIG_URL_PREFIX="https://packagecloud.io/netdata/netdata-repoconfig/packages"
+REPOCONFIG_VERSION="1-1"
+PATH="${PATH}:/usr/local/bin:/usr/local/sbin"
+
+# ======================================================================
+# Defaults for environment variables
+
+RELEASE_CHANNEL="nightly"
+NETDATA_CLAIM_URL="https://app.netdata.cloud"
+NETDATA_USER_CONFIG_DIR="/etc/netdata"
+NETDATA_AUTO_UPDATES="1"
+NETDATA_ONLY_NATIVE=0
+NETDATA_ONLY_STATIC=0
+
+NETDATA_DISABLE_TELEMETRY="${DO_NOT_TRACK:-0}"
+NETDATA_TARBALL_BASEURL="${NETDATA_TARBALL_BASEURL:-https://storage.googleapis.com/netdata-nightlies}"
+
+if [ ! -t 1 ]; then
+  INTERACTIVE=0
+else
+  INTERACTIVE=1
+fi
+
+# ======================================================================
+# Usage info
+
+usage() {
+  cat << HEREDOC
+USAGE: kickstart.sh [options]
+       where options include:
+
+  --dont-wait                Do not prompt for user input. (default: prompt if there is a controlling terminal)
+  --non-interactive          Do not prompt for user input.
+  --dont-start-it            Do not start the agent by default (only for static installs or local builds)
+  --interactive              Prompt for user input even if there is no controlling terminal.
+  --stable-channel           Install a stable version instead of a nightly build (default: install a nightly build)
+  --nightly-channel          Install a nightly build instead of a stable version
+  --no-updates               Do not enable automatic updates (default: enable automatic updates)
+  --auto-update              Enable automatic updates.
+  --disable-telemetry        Opt-out of anonymous statistics.
+  --native-only              Only install if native binary packages are available.
+  --static-only              Only install if a static build is available.
```

Comment:
I think we should keep only the `--native-only` option to prioritize binary packages. There's no reason to provide the `--static-only` option as it is the default fallback when there's no native package. If we need it internally we can add a hidden environment flag for this.

Reply:
The intent here is to have an option to explicitly ask for certain build types. There will also be a `--build-only` to explicitly ask for building locally.

The main reason for this is that there is _not_ perfect feature parity between install methods (for example, eBPF does not work right now in the static builds, and users can’t use a custom install prefix with anything other than a local build), so there are practical reasons to want a specific install type (aside from the obvious case of explicitly testing particular install types in particular environments where they would not normally be the default).

Reply:
> eBPF does not work right now in the static builds
> users can’t use a custom install prefix with anything other than a local build

Those are limitations that come from the build process. Is there anything that forbids us from (a) making eBPF work correctly, and (b) using a custom installation prefix, for static builds?

> there are practical reasons to want a specific install type (aside from the obvious case of explicitly testing particular install types in particular environments where they would not normally be the default).

I'm interested in finding an example that requires these options to be explicitly available to the end users. I think testing is not relevant in this case because we can always do this internally with an environment variable or option that's not displayed in the help text.

> The main reason for this is that there is not perfect feature parity between install methods

Can you expand on this? I think, I understand the case where we might not be able to install the same set of external collectors. I'm curious, if the feature parity concerns the Netdata binary itself.